tests/sca/test_sca.py

```diff
@@ -0,0 +1,170 @@
+        """Test the hypervisor facts in mapping
+
+        :title: virt-who: sca: test hypervisor facts
+        :id: d0ca112a-6965-4334-8b41-410bb4c74e0c
+        :caseimportance: High
+        :tags: tier1
+        :customerscenario: false
+        :upstream: no
+        :steps:
+            1. Run virt-who to get the mapping
+            2. Check the hypervisor facts
+        :expectedresults:
+            1. virt-who can get and send maping successfully
+            2. the hypervisor facts of type, version, socket, dmi and cluster
+                are expected.
+        """
+        register_org = register_data['default_org']
+        hypervisor_hostname = hypervisor_data['hypervisor_hostname']
+        result = virtwho.run_cli()
+        assert (result['send'] == 1
+                and result['error'] == 0)
+
+        facts = result['mappings'][register_org][hypervisor_hostname]
+        assert (facts['type'] == hypervisor_data['type']
+                and facts['version'] == hypervisor_data['version']
+                and facts['socket'] == hypervisor_data['cpu']
+                and facts['dmi'] == hypervisor_data['hypervisor_uuid'])
+        if HYPERVISOR in ['esx', 'rhevm', 'ahv']:
+            assert facts['cluster'] == hypervisor_data['cluster']
+
+    def test_host_to_guest_association(self, virtwho, sm_guest, ssh_guest,
+                                       register_data, hypervisor_data,
+                                       register_guest):
+        """Test the host-to-guest association in mapping log and register server
+        Web UI.
+
+        :title: virt-who: sca: test host-to-guest association
+        :id: 044a7c1b-92cb-43a0-a790-f81279e09a8b
+        :caseimportance: High
+        :tags: tier1
+        :customerscenario: false
+        :upstream: no
+        :steps:
+            1. run virt-who against Satellite/RHSM
+            2. assert the association in mapping
+            3. assert the association in Satellite/RHSM WebUI
+        :expectedresults:
+            1. the host and guest are associated in the both mapping
+            and register server WebUI
+        """
+        guest_uuid = hypervisor_data['guest_uuid']
+        guest_hostname = hypervisor_data['guest_hostname']
+        hypervisor_hostname = hypervisor_data['hypervisor_hostname']
+        default_org = register_data['default_org']
+        # assert the association in mapping
+        result = virtwho.run_cli()
+        mappings = result['mappings']
+        associated_hypervisor_in_mapping = mappings[default_org][guest_uuid][
+            'guest_hypervisor']
+        assert (result['send'] == 1
+                and result['error'] == 0
+                and associated_hypervisor_in_mapping == hypervisor_hostname)
+        # assert the association in Satellite web
+        if REGISTER == 'satellite':
+            assert register.associate(hypervisor_hostname,
+                                      guest_hostname) is not False
```

Comment:
the default will pass if not False/None, same as
```
            assert register.associate(hypervisor_hostname,
                                      guest_hostname)
```

Reply:
Updated.